web/models/user.ex

```diff
@@ -0,0 +1,36 @@
+defmodule App.User do
+  @moduledoc false
+
+  use App.Web, :model
+
+  alias Comeonin.Bcrypt
+
+  schema "users" do
+    field :username, :string
+    field :password, :string, virtual: true
+    field :password_hash, :string
+  end
+
+  def changeset(model, params \\ :invalid) do
+    model
+    |> cast(params, ~w[username])
+    |> validate_required([:username])
```

Comment:
validate_required(:username) can take just atoms instead of a list: https://hexdocs.pm/ecto/Ecto.Changeset.html#validate_required/3-examples